core/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridParser.java

```diff
@@ -77,12 +74,14 @@ public AggregatorFactory parse(String aggregationName, XContentParser parser, Se
                 currentFieldName = parser.currentName();
             } else if (vsParser.token(currentFieldName, token, parser)) {
                 continue;
-            } else if (token == XContentParser.Token.VALUE_NUMBER) {
-                if ("precision".equals(currentFieldName)) {
-                    precision = parser.intValue();
-                } else if ("size".equals(currentFieldName)) {
+            } else if (token == XContentParser.Token.VALUE_NUMBER ||
+                    token == XContentParser.Token.VALUE_STRING) { //Be lenient and also allow numbers enclosed in quotes
+                if (GeoHashGridParams.FIELD_PRECISION.equals(currentFieldName)) {
+                    precision = GeoHashGridParams.checkPrecision(parser.intValue());
+                } else if (GeoHashGridParams.FIELD_SIZE.equals(currentFieldName)) {
                     requiredSize = parser.intValue();
-                } else if ("shard_size".equals(currentFieldName) || "shardSize".equals(currentFieldName)) {
+                } else if (GeoHashGridParams.FIELD_SHARD_SIZE.equals(currentFieldName) ||
+                        GeoHashGridParams.FIELD_SHARD_SIZE_FALLBACK.equals(currentFieldName)) {
                     shardSize = parser.intValue();
                 }
             }
```

Comment:
I think part of the bug is also that we don't have an `else` statement here where we throw an exception. Which means that if you provide a parameter with eg. a boolean value (for which we don't check above), it will be silently ignored. Can you add such an `else` statement (you can look eg. at MissingParser for an example).